sdk/azcore/arm/runtime/pipeline.go

```diff
@@ -7,36 +7,62 @@
 package runtime
 
 import (
+	"fmt"
+	"reflect"
+
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
 	armpolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/pipeline"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/shared"
+	azpolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	azruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 )
 
 // NewPipeline creates a pipeline from connection options.
 // The telemetry policy, when enabled, will use the specified module and version info.
+// This method panics when the ClientOptions.Cloud field is set with a Configuration object
+// that's missing Azure Resource Manager settings. A future version will return an error instead.
 func NewPipeline(module, version string, cred shared.TokenCredential, plOpts azruntime.PipelineOptions, options *arm.ClientOptions) pipeline.Pipeline {
 	if options == nil {
 		options = &arm.ClientOptions{}
 	}
-	ep := options.Endpoint
-	if len(ep) == 0 {
-		ep = arm.AzurePublicCloud
+	conf, err := getConfiguration(&options.ClientOptions)
+	if err != nil {
+		// TODO: return the error
+		panic(err)
+	}
+	scope := conf.Audiences[0]
```

Comment:
Do we need to doc that the first audience is the scope?  In what scenarios will more than one audience be specified?

Reply:
Thanks for commenting on this line, there's a bug here (scope = audience + "/.default"). 

AAD requires exactly one audience per token. Arguably we should have `Audience string` instead (item for future discussion). However, ARM metadata contains two audiences. I believe this is for historical reasons. One audience is for Service Management, the other for Resource Manager. The latter accepts tokens for the former, so I don't know any functional reason to prefer one over the other and don't think we need to document our arbitrary choice.